### ISSUE TITLE
ERROR: dependency ‘rlist’ is not available for package ‘highcharter’

### ISSUE THREAD

**Issue by Tartomas** [jbkunst/highcharter#645]
Hello, 
I have a Docker file to work with a Shiny app in ShinyProxy. I'm using a r-base-3.6.2 from this [Docker Hub](https://hub.docker.com/r/openanalytics/r-base/tags)

Since a couples of days a I'm facing this error and I don't know  how to resolve it. 
Please help !

> docker build -t highcharter .

```yml
FROM openanalytics/r-base:latest

# system libraries of general use
RUN apt-get update && apt-get install -y \
    libcurl4-gnutls-dev \
    libcairo2-dev \
    libxt-dev \
    libssl-dev \
    libssh2-1-dev \ 
    libssl1.0.0

# system library dependency for the 
RUN apt-get update && apt-get install -y \
    libmpfr-dev \
    libxml2-dev 

RUN R -e "install.packages(c('shiny', 'rmarkdown'), repos='https://cloud.r-project.org/')"
# RUN R -e "install.packages(c('tidyverse'), repos='https://cloud.r-project.org/')"
RUN R -e "install.packages(c('highcharter'))"  

```
```r
ERROR: dependency ‘rlist’ is not available for package ‘highcharter’
* removing ‘/usr/local/lib/R/site-library/highcharter’

The downloaded source packages are in
	‘/tmp/RtmpUKgW2N/downloaded_packages’
Warning messages:
1: In install.packages(c("highcharter")) :
  installation of package ‘rlist’ had non-zero exit status
2: In install.packages(c("highcharter")) :
  installation of package ‘highcharter’ had non-zero exit status

```

Then, when I run the image to try it anf this happends: 
> docker run -i -t highcharter 

```r

R version 3.6.2 (2019-12-12) -- "Dark and Stormy Night"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages("rlist")
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Warning: dependency ‘XML’ is not available
trying URL 'https://cloud.r-project.org/src/contrib/rlist_0.4.6.1.tar.gz'
Content type 'application/x-gzip' length 71681 bytes (70 KB)
==================================================
downloaded 70 KB

ERROR: dependency ‘XML’ is not available for package ‘rlist’
* removing ‘/usr/local/lib/R/site-library/rlist’

The downloaded source packages are in
	‘/tmp/Rtmpt0NX8b/downloaded_packages’
Warning message:
In install.packages("rlist") :
  installation of package ‘rlist’ had non-zero exit status
> install.packages("XML")
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Warning message:
package ‘XML’ is not available (for R version 3.6.2) 
```

**Comment by pachamaltese** [comment]
Hi
The fact that XML is not available suggests that there is a missing APT package.
Check this for example https://github.com/pachamaltese/rstudio-server-highcharter-droplet/blob/master/scripts/01-packages.sh#L20
You can edit the docker image to follow the steps from the script and both shiny and highcharter will work.

**Comment by Tartomas** [comment]
Hi Pacha!, I will trying your option. I tell you how it goes.

**Comment by Tartomas** [comment]
Hi @pachamaltese 
Nope, same problem with rlist
This is the Docker file 

  ```bash
FROM openanalytics/r-base

# system libraries of general use
RUN apt-get update && apt-get install -y \
    libcurl4-gnutls-dev \
    libcairo2-dev \
    libxt-dev \
    libssl-dev \
    libssh2-1-dev \ 
    libssl1.0.0

# system library dependency for the 
RUN apt-get update && apt-get install -y \
    libmpfr-dev \
    libxml2-dev 

RUN apt-get -y install libxml2-dev libssl-dev libcurl4-openssl-dev

RUN R -e "install.packages(c('shiny', 'rmarkdown'), repos='https://cloud.r-project.org/')"

RUN R -e "install.packages(c('highcharter'))"  

EXPOSE 3838
  ```

**Comment by Tartomas** [comment]
Hello I resolved it by updating to R 4.0, everythings goes smooth now.
